src/alg/rsa.rs

```diff
@@ -162,13 +162,88 @@ impl std::error::Error for ModulusBitsError {}
 /// [RSA]: https://en.wikipedia.org/wiki/RSA_(cryptosystem)
 /// [RFC 7518]: https://www.rfc-editor.org/rfc/rfc7518.html
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-#[cfg_attr(docsrs, doc(cfg(feature = "rsa")))]
+#[cfg_attr(docsrs, doc(cfg(feature = ["rsa", "rsa-byo"])))]
 pub struct Rsa {
     hash_alg: HashAlg,
     padding_alg: Padding,
 }
 
-impl Algorithm for Rsa {
+/// Integrity algorithm using [RSA] digital signatures with self-brought RNG.
+///
+/// Depending on the variation, the algorithm employs PKCS#1 v1.5 or PSS padding and
+/// one of the hash functions from the SHA-2 family: SHA-256, SHA-384, or SHA-512.
+/// See [RFC 7518] for more details. Depending on the chosen parameters,
+/// the name of the algorithm is one of `RS256`, `RS384`, `RS512`, `PS256`, `PS384`, `PS512`:
+///
+/// - `R` / `P` denote the padding scheme: PKCS#1 v1.5 for `R`, PSS for `P`
+/// - `256` / `384` / `512` denote the hash function
+///
+/// The length of RSA keys is not unequivocally specified by the algorithm; nevertheless,
+/// it **MUST** be at least 2048 bits as per RFC 7518. To minimize risks of misconfiguration,
+/// use [`StrongAlg`](super::StrongAlg) wrapper around `Rsa`:
+///
+/// ```
+/// # use jwt_compact::alg::{StrongAlg, Rsa};
+/// let mut rng = OsRng;
+/// let cell = RefCell::new(&mut rng);
+/// const ALG: StrongAlg<Rsa> = StrongAlg(Rsa::rs256().with_byo(cell));
+/// // `ALG` will not support RSA keys with unsecure lengths by design!
+/// ```
+///
+/// [RSA]: https://en.wikipedia.org/wiki/RSA_(cryptosystem)
+/// [RFC 7518]: https://www.rfc-editor.org/rfc/rfc7518.html
+#[derive(Debug, PartialEq, Eq)]
+#[cfg_attr(docsrs, doc(cfg(feature = "rsa-byo")))]
+#[cfg(feature = "rsa-byo")]
+pub struct RsaByo<'a, Rng: CryptoRngCore> {
+    hash_alg: HashAlg,
+    padding_alg: Padding,
+
+    rng: RefCell<&'a mut Rng>,
```

Comment:
A `RefCell` might not be the cleanest solution. I was struggling very much to get a `&mut impl CryptoRngCore` into `sign` without modifying the signature into a `&mut self` and breaking everything. I'm very much open to other solutions